src/jit/flowgraph.cpp

```diff
@@ -20245,14 +20257,14 @@ PRED_OK:;
         bool compThisArgAddrExposedOK = !lvaTable[info.compThisArg].lvAddrExposed;
 #ifndef JIT32_GCENCODER
         compThisArgAddrExposedOK = compThisArgAddrExposedOK || copiedForGenericsCtxt;
-#endif  // !JIT32_GCENCODER
-        noway_assert(compThisArgAddrExposedOK &&                  //     should never expose the address of arg 0 or
-                     !lvaTable[info.compThisArg].lvArgWrite &&    //     write to arg 0.
-                     (                                                  //   In addition,
-                        lvaArg0Var == info.compThisArg ||               //     lvArg0Var should remain 0 if arg0 is not written to or address-exposed.
-                        lvaArg0Var != info.compThisArg &&
-                        (lvaTable[lvaArg0Var].lvAddrExposed || lvaTable[lvaArg0Var].lvArgWrite || copiedForGenericsCtxt)
-                     ));
+#endif                                                         // !JIT32_GCENCODER
+        noway_assert(compThisArgAddrExposedOK &&               //     should never expose the address of arg 0 or
```

Comment:
Can we move the comment on this assert to be above it? The way it is currently written is super confusing